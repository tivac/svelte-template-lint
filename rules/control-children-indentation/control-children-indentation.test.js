"use strict";

const dedent = require("dedent");

const lint = require("../../index.js");

const config = (indent) => ({
    rules : {
        "control-children-indentation" : [ "warn", {
            indent,
        }],
    },
});

const specimens = [
    [
        "indent 4 success",
        `
        {#if true}
            <div>a</div>
        {/if}
        `,
        config(4),
    ],
    [
        "indent 4 failure",
        `
        {#if true}
        <div>a</div>
        {/if}
        `,
        config(4),
    ],
    [
        "indent 0 success",
        `
        {#if true}
        <div>a</div>
        {/if}
        `,
        config(0),
    ],
    [
        "indent 0 failure",
        `
        {#if true}
            <div>a</div>
        {/if}
        `,
        config(0),
    ],
];

it.each(
    specimens.map(([ name, spec, options ]) => [ name, dedent(spec), options ])
)("control-children-indentation - %s", async (name, spec, options) => {
    const result = await lint(`${name}.svelte`, spec, options);

    // Remove node prop from returned messages, makes snapshots too chatty
    const pruned = result.map(({ node, ...rest }) => rest);

    expect(pruned).toMatchSnapshot();
});

