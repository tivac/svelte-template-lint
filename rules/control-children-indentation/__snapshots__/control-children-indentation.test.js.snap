// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`control-children-indentation - indent 0 failure 1`] = `
Array [
  Object {
    "message": "Expected indentation of 0, got 4 instead",
    "plugin": "control-children-indentation",
  },
]
`;

exports[`control-children-indentation - indent 0 success 1`] = `Array []`;

exports[`control-children-indentation - indent 4 failure 1`] = `
Array [
  Object {
    "message": "Expected indentation of 4, got 0 instead",
    "plugin": "control-children-indentation",
  },
]
`;

exports[`control-children-indentation - indent 4 success 1`] = `Array []`;
